src/lib/rest/restReply.cpp

```diff
@@ -117,6 +117,26 @@ void restReply(ConnectionInfo* ciP, const std::string& answer)
     }
   }
 
+  if (strlen(restCORSEnabled) > 0)
+  {
+    // If any origin is allowed, the header is sent always with "any" as value
+    if (strcmp(restCORSEnabled, "__ALL") == 0)
+    {
+      MHD_add_response_header(response, "Access-Control-Allow-Origin", "*");
+    }
+    // If a specific origin is allowed, the header is only sent if the origins match
+    else if (strcmp(ciP->httpHeaders.origin.c_str(), restCORSEnabled) == 0)
+    {
+      MHD_add_response_header(response, "Access-Control-Allow-Origin", restCORSEnabled);
+    }
+
+    if (ciP->verb == OPTIONS)
+    {
+      MHD_add_response_header(response, "Access-Control-Allow-Headers", "User-Agent, Fiware-Service, Fiware-Servicepath, Ngsiv2-AttrsFormat, Fiware-Correlator, X-Forwarded-For, X-Real-IP, X-Auth-Token");
+      MHD_add_response_header(response, "Access-Control-Max-Age", "86400");
```

Comment:
How long is 86400? Why is this hardwired?

Reply:
It is in seconds, corresponding to 24 hours. This value is the cap for Firefox but is different for other browsers (10 minutes for Chrome for example). 

This is hardwired at the moment since we planned this issue to be handled in several PRs, I thought we could discuss and decide on how to handle it in the latter PRs. I think we should pass it as a parameter to the switch. I am not sure about the allowed headers though.

Reply:
> I think we should pass it [the max age] as a parameter to the switch

I agree.

In the meanwhile, if you don't want to create that switch for this PR, add a "FIXME" comment about it just above this line.

Reply:
Fixed in 8c48f60